### ISSUE TITLE
Community provider > Service type input is not updating aria-expanded correctly

### ISSUE THREAD

**Issue by qjac** [department-of-veterans-affairs/va.gov-cms#22926]
## Describe the defect
**Note: this ticket should be completed before #22830** because it will make it easier to implement and test if this is done first.

aria-expanded notifies assitive tech when a dropdown is expanded or collapsed. This usually triggers a screenreader announcement to let screenreader users know when a dropdown is open or closed. The Community provider > Service type field dropdown is usually correct in matching the aria-expanded state to the actual state of the dropdown, but sometimes there’s a mismatch. When a single character is typed into the input, aria-expanded changes to true, but the dropdown is still closed at that point. This mismatch means that screenreader users cannot reliably know when the dropdown is truly expanded or collapsed.

 - **[Severity level](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric):** 1
- **[Experience standard](https://design.va.gov/about/experience-standards/):** User can perceive all meaningful elements, User can identify an element and its state

## To Reproduce
1. Go to Facility Locator - Community Provider Facility Type 
2. Open devtools.
3. Tab or click into the field.
4. Type a character and note that the dropdown is closed, but aria-expanded=true
5. Type a second character. Note the dropdown opens, aria-expanded=true (so now it's correct).
6. Delete a character. Note that the dropdown closes, but aria-expanded=true. Wrong again.
7. Also try a screenreader while opening and closing the dropdown.

## Screenshots
Community provider > Service type dropdown is closed, but aria-expanded is set to true. 😳 This is after typing a single character into the combobox.

<img width="994" height="197" alt="Image" src="https://github.com/user-attachments/assets/da1b8aac-9835-4b5a-b42d-94ea4a5d8117" />

## Recommendation
Ensure aria-expanded is true only when the dropdown is expanded and is false when dropdown is collapsed.

## Additional notes
- 

## AC / Expected behavior
For the Community provider > Service type input,
- [ ] aria-expanded=true when the dropdown is open
- [ ] aria-expanded=false when the dropdown is closed

## Definition of done
- [x] Code completed; PR created
- [x] Review instance created/available; internal team notified to start reviews
- [x] Code review (meets established quality and security standards)
- [ ] ~Design review (meets design & UX requirements)~
- [x] A11y review (meets accessibility standards) REQUIRED
- [x] Functional/feature review (meets acceptance criteria)
- [x] Automated tests passed; PR approved
- [x] Testing completed; dev team notified to deploy
- [x] PO/PM/external verify in prod
- [ ] Documentation updated

**Comment by carlyMhayter** [comment]
So interestingly in staging, I do not see this issue! There also appears to be some other aria attributes on the input, such as aria-controls, which are present in staging but not in the screenshot you posted @qjac. I had started work on this because when I **first** pulled from main and started local development, I DID see the issue you described...but I just pulled from main again and I no longer see this issue in local development either. 

Because I cannot replicate this issue in staging, I would consider this closed? But I'll leave the final check up to you

<img width="1126" height="580" alt="Image" src="https://github.com/user-attachments/assets/0e375da1-9c73-4ae9-9887-d4b3c56554ec" />
<img width="1109" height="592" alt="Image" src="https://github.com/user-attachments/assets/46033537-ae70-4d2f-aadc-6b7e1e043dad" />

**Comment by r-bartlett-gsa** [comment]
@carlyMhayter I attempted to test this on this RI: http://9dde6e802bfa40ff3c60e08421a66281.vfs.va.gov/ and as soon as I select Community providers for the facility type the entire facility locator disappears.

**Comment by qjac** [comment]
@r-bartlett-gsa @sky3an1co, the link to the  review instance isn't loading for me. Does it need to be rebuilt?
